movement/io/save_poses.py

```diff
@@ -11,72 +11,133 @@
+    'split_individuals' argument.
 
     Parameters
     ----------
     ds : xarray Dataset
         Dataset containing pose tracks, confidence scores, and metadata.
+    split_individuals : bool, optional
+        If True, return a dictionary of pandas DataFrames, for each individual.
+        If False, return a single pandas DataFrame with multi-index columns
+        for all individuals.
+        Default is True.
 
     Returns
     -------
-    pandas DataFrame
+    pandas DataFrame or dict
+        DeepLabCut-style pandas DataFrame or dictionary of DataFrames.
 
     Notes
     -----
-    The DataFrame will have a multi-index column with the following levels:
-    "scorer", "individuals", "bodyparts", "coords" (even if there is only
-    one individual present). Regardless of the provenance of the
-    points-wise confidence scores, they will be referred to as
-    "likelihood", and stored in the "coords" level (as DeepLabCut expects).
+    The DataFrame(s) will have a multi-index column with the following levels:
+    "scorer", "individuals", "bodyparts", "coords"
+    (if multi_individual is True),
+    or "scorer", "bodyparts", "coords" (if multi_individual is False).
+    Regardless of the provenance of the points-wise confidence scores,
+    they will be referred to as "likelihood", and stored in
+    the "coords" level (as DeepLabCut expects).
 
     See Also
     --------
     to_dlc_file : Save the xarray dataset containing pose tracks directly
         to a DeepLabCut-style ".h5" or ".csv" file.
     """
-
     if not isinstance(ds, xr.Dataset):
         error_msg = f"Expected an xarray Dataset, but got {type(ds)}. "
         logger.error(error_msg)
         raise ValueError(error_msg)
 
     ds.poses.validate()  # validate the dataset
 
-    # Concatenate the pose tracks and confidence scores into one array
-    tracks_with_scores = np.concatenate(
-        (
-            ds.pose_tracks.data,
-            ds.confidence.data[..., np.newaxis],
-        ),
-        axis=-1,
-    )
-
-    # Create the DLC-style multi-index columns
-    # Use the DLC terminology: scorer, individuals, bodyparts, coords
     scorer = ["movement"]
-    individuals = ds.coords["individuals"].data.tolist()
     bodyparts = ds.coords["keypoints"].data.tolist()
-    # The confidence scores in DLC are referred to as "likelihood"
     coords = ds.coords["space"].data.tolist() + ["likelihood"]
 
-    index_levels = ["scorer", "individuals", "bodyparts", "coords"]
-    columns = pd.MultiIndex.from_product(
-        [scorer, individuals, bodyparts, coords], names=index_levels
-    )
-    df = pd.DataFrame(
-        data=tracks_with_scores.reshape(ds.dims["time"], -1),
-        index=np.arange(ds.dims["time"], dtype=int),
-        columns=columns,
-        dtype=float,
-    )
-    logger.info("Converted PoseTracks dataset to DLC-style DataFrame.")
-    return df
-
-
-def to_dlc_file(ds: xr.Dataset, file_path: Union[str, Path]) -> None:
+    if split_individuals:
```

Comment:
Hi @niksirbi, 

If split_individuals == True and we have only one individual, it will just output one single-animal DataFrame (with "scorer", "bodyparts", "coords"). If split_individuals == False and we have only one individual, it will output one single-animal DataFrame (with "scorer", "individuals", "bodyparts", "coords"). 

Although for the second case when split_individuals == False the individuals column will just be filled with one individual, it might still be important to have this feature. It might be useful in case they wanted to merge the dataframes with other multi-individual dataframes, where pandas would want the dataframes to have the same format. 

It also might be unexpected if they ran the function with split_individuals == False on a set of data with a mixture of single and multi-individual xarrays and saw the output being a mixture of single-animal and multi-animal dataframes as the single-individual xarrays would automatically turn into single-individual dataframes with no choice to make it multi-individual. 

The auto function in to_dlc_file handles cases when the user might want all single individual xarrays to be stored as single-individual dataframes and multi-individual xarrays to be stored as multi-individual dataframes, I can make a separate function for this auto feature and also use it for to_dlc_df if preferable.

Reply:
> If split_individuals == True and we have only one individual, it will just output one single-animal DataFrame (with "scorer", "bodyparts", "coords"). If split_individuals == False and we have only one individual, it will output one single-animal DataFrame (with "scorer", "individuals", "bodyparts", "coords").

Hm, I actually like this suggestion and the flexibility in gives to the user. The way you are proposing means that `split_individuals == True` will always give an output `csv` in "single-animal" format, while `split_individuals == False` will always return a csv in "multi-animal" format, regardless of how many animals are in the project.

The arguments you make for it are convincing, so let's go ahead and do this! We just have to be careful to write the docstrings in an understandable way.